src/main/java/it/gov/pagopa/reward/service/lookup/HpanInitiativeMediatorServiceImpl.java

```diff
@@ -136,23 +149,50 @@ private Flux<HpanUpdateEvaluateDTO> initializingHpanInitiativeDTO(HpanInitiative
     }
 
     private Mono<String> findAndModify(HpanUpdateEvaluateDTO hpanUpdateEvaluateDTO) {
+
         return hpanInitiativesRepository.findById(hpanUpdateEvaluateDTO.getHpan())
                 .switchIfEmpty(Mono.defer(() -> getNewHpanInitiatives(hpanUpdateEvaluateDTO)))
                 .mapNotNull(hpanInitiatives -> hpanInitiativesService.evaluate(hpanUpdateEvaluateDTO, hpanInitiatives))
+                .flatMap(oi -> initializeCounterAndIfRetrieveFamily(hpanUpdateEvaluateDTO, oi))
                 .flatMap(oi -> hpanInitiativesRepository.setInitiative(hpanUpdateEvaluateDTO.getHpan(), oi))
                 .map(ur -> hpanUpdateEvaluateDTO.getHpan());
     }
 
+    @NotNull
+    private Mono<OnboardedInitiative> initializeCounterAndIfRetrieveFamily(HpanUpdateEvaluateDTO hpanUpdateEvaluateDTO, OnboardedInitiative oi) {
+        if(HpanInitiativeConstants.OPERATION_ADD_INSTRUMENT.equals(hpanUpdateEvaluateDTO.getOperationType()) && oi.getActiveTimeIntervals().size() == 1){
+            return retrieveAndEvaluateInitiative(hpanUpdateEvaluateDTO)
+                    .doOnNext(oi::setFamilyId)
+                    .switchIfEmpty(Mono.just(hpanUpdateEvaluateDTO.getUserId()))
+                    .flatMap(userFamilyId -> userInitiativeCountersRepository.createIfNotExists(userFamilyId, hpanUpdateEvaluateDTO.getInitiativeId()))
```

Comment:
it's too ambigous as name
```suggestion
                    .flatMap(counterSubjectId -> userInitiativeCountersRepository.createIfNotExists(counterSubjectId , hpanUpdateEvaluateDTO.getInitiativeId()))
```